dashboard.el

```diff
@@ -267,6 +267,36 @@ example:
     (when (and items-pg (< items-id items-len))
       (dashboard--goto-line items-pg))))
 
+(defun dashboard-cycle-section-forward (&optional section)
```

Comment:
I, for example, have set it instead of the jump between sections.
I'm not a programmer, I study and write papers so I have a lot of files in my recents and bookmarks I jump between, and found dashboard to be a nice visualisation of all this.
Since going in the recents and then TAB-ing all the way down 19 files, or having to calculate if going in the bookmarks and shift-tabbing is quicker, I just find more natural to jump and advance with the same keybinding.

I get this hasn't been requested, so it's hard to know if it's because they haven't thought of that or they really wouldn't have use for it, so I implemented them as to not touch old user's spaces and configs.
I also get you not wanting to put everyone's particular use case in the package, but a "local" cycling function sounds like one of the first things that would have been implemented, just after forwards, backwards and gotos - at least to me.

Reply:
Ok, there's no problem in adding new features, I just want to know how people are using it. A couple of questions, how are you binding the function to the keybinding? Do you have two keybindings, one for bookmarks and another for recent files?

We have [ace-link-dashboard](https://github.com/emacs-dashboard/ace-link-dashboard) that maybe work better to jump directly to the file (widget) you want.

If you still want this feature, I think a better approach would be:
```emacs-lisp
(defun dashboard-cycle-section-forward (&optional section)
  (if section 
    (progn
      (ignore-errors (dashboard--goto-section section))
      (widget-forward 1))
    (widget-forward 1))
```
But, I don't know if this code fulfills your needs, does it?

Reply:
My functions act as widget-forward and widget-backward, but ensuring the cycling within the specified or current section.

I have a rough implementation for development purposes as:

```
;; Custom keybindings
(define-key dashboard-mode-map (kbd "r") (lambda () (interactive) (dashboard-cycle-section-forward 'recents)))
(define-key dashboard-mode-map (kbd "R") (lambda () (interactive) (dashboard-cycle-section-backward 'recents)))
(define-key dashboard-mode-map (kbd "m") (lambda () (interactive) (dashboard-cycle-section-forward 'bookmarks)))
(define-key dashboard-mode-map (kbd "M") (lambda () (interactive) (dashboard-cycle-section-backward 'bookmarks)))
;; Widget navigation
(define-key dashboard-mode-map (kbd "n") (lambda () (interactive) (widget-forward 1)))
(define-key dashboard-mode-map (kbd "p") (lambda () (interactive) (widget-backward 1)))
```

Which allows me whole dashboard navigation with n and p, while I can navigate, with cycling, only the recents or only the bookmarks with r (and R) and m (or M).

Cool thing a friend requested was current section navigation as default, so that, for example:

`(define-key dashboard-mode-map (kbd "i") (lambda () (interactive) (dashboard-cycle-section-forward)))`

Allows him to go forward and only cycle the current section without need for specifications.

I had thought of implementing it with some `dashboard-navigation-cycle-current-section` variable, which we can still do for easier setting, but specific section cycling, like my config, would still need to be set and specified, so that added quite some complexity for little gain, at the cost of maintenance efforts.